src/include/daos_uns.h

```diff
@@ -187,6 +177,32 @@ duns_destroy_path(daos_handle_t poh, const char *path);
 int
 duns_parse_attr(char *str, daos_size_t len, struct duns_attr_t *attr);
 
+/**
+ * Set the pool label in the duns attr struct in case it was fetched in a different way than
+ * duns_resolve_path(). This is used in case user wants to call duns_destroy_attr() to free that
+ * buffer rather than free those independently.
```

Comment:
no the ides is to use this function in conjunction with duns_destroy_attr. we are using D_STRNDUP in there.
if use want to free that buffer themselves, they don't use this function.
this is all about memtracing

Reply:
Ah, so instead of this:
```
attrp->da_cont_label = strdup(my_label)
...
free(attrp->da_cont_label);
duns_destroy_attr(attrp);
```
The user can now do this:
```
duns_set_pool_label(attrp, my_label);
duns_destroy_attr(attrp);
```

Reply:
exactly